Saved Tones, Multi Tone Paging, UI Redesign

DeathSoulYTDev opened the pull request:
- Added Support For Multi Tone Paging

DeathSoulYTDev committed:
Add files via upload

DeathSoulYTDev committed:
Add files via upload

DeathSoulYTDev committed:
Update codeplug.yml

DeathSoulYTDev committed:
Add files via upload

DeathSoulYTDev committed:
Added Multi Tone Paging - QC 2

DeathSoulYTDev committed:
Tones Delay

- Removed Delay for First Tone When Using Stacked Tones
- Lowered the Delay from 2000ms to 1750ms

DeathSoulYTDev committed:
Zone Tabs

- Loaded Zones and there channels into there own tabs
- Added a Zones Filter to QC2 Tones to allow setting Tones Per Zone/Agency

DeathSoulYTDev committed:
Updated WLLib

firealarmss commented on WhackerLinkConsoleV2/App.xaml.cs:
```diff
@@ -9,6 +10,20 @@ namespace WhackerLinkConsoleV2
     /// </summary>
     public partial class App : Application
     {
+        public Codeplug Codeplug { get; set; }
```
Why is all this changed?

firealarmss commented on WhackerLinkConsoleV2/ChannelBox.xaml.cs:
```diff
@@ -300,6 +300,7 @@ private async void PTTButton_Click(object sender, RoutedEventArgs e)
         private void PageSelectButton_Click(object sender, RoutedEventArgs e)
         {
             if (!IsSelected) return;
+            
```
Blank line added?

firealarmss commented on WhackerLinkConsoleV2/MainWindow.xaml.cs:
```diff
@@ -166,255 +203,340 @@ private void LoadCodeplug(string filePath)
             }
         }
 
+        //private void GenerateChannelWidgets()
```
What is this about?

firealarmss commented on WhackerLinkConsoleV2/MainWindow.xaml.cs:
```diff
@@ -435,19 +557,374 @@ private void GenerateChannelWidgets()
+        }
+
+
+        //private void AddToneSets(Canvas zoneCanvas, double offsetX, double offsetY, string zone)
```
What is this about?

firealarmss commented on WhackerLinkConsoleV2/MainWindow.xaml.cs:
```diff
@@ -2314,5 +2956,10 @@ private void CallHist_Click(object sender, RoutedEventArgs e)
             callHistoryWindow.Show();
         }
+
+        private void MenuItem_Click(object sender, RoutedEventArgs e)
```
why?

DeathSoulYTDev committed:
Comments Removal & Tone Delay

- Removed All Unused/Old Code that was Commented Out
- Fixed the Stacked Tones Delay so its not delaying the First Tone

DeathSoulYTDev committed:
UI Fix

- Fixed Canvas Sizing When Window Resizes
- Auto Added A SYSTEMS Tab for Show a Playback Channel and SystemBoxes

DeathSoulYTDev committed:
Comments Removal

- Removed Forgotten Commented out Code

DeathSoulYTDev commented:
i have Updated the the code and removed all unneeded code that was commented out i have also fixed the UI Bug with tabs

DeathSoulYTDev commented:
This has also been tested with the fivem script and I have confirmed that it works

DeathSoulYTDev committed:
Update/Bug Fixes

- Fixed The Global Ptt & Alert Tone 3 Buttons as they where previously swaped Somehow
- Updated Styling Of the Channels Groups/ZoneTabs
- Updated Styling to a Dark Mode

DeathSoulYTDev committed:
Update README.md

DeathSoulYTDev committed:
Update README.md

DeathSoulYTDev committed:
Update README.md

DeathSoulYTDev committed:
Update image in README for WhackerLink Console V2

Firav commented:
@firealarmss My suggestion would be to close this PR and #7.
@DeathSoulYTDev We need separate PR's made for the changed you're making. The changes outlined in this PR are not just adding support for multi-tone paging, but also large overhauls to the Console UI itself. Please either isolate the changes into different PR's or open a new PR with all changes listed so that we can properly review the changes you're proposing without having to cross-reference your #7 as well. It also seems like there were some changes requested in #7 were not addressed in this PR as well.

DeathSoulYTDev commented:
> @firealarmss My suggestion would be to close this PR and #7. @DeathSoulYTDev We need separate PR's made for the changed you're making. The changes outlined in this PR are not just adding support for multi-tone paging, but also large overhauls to the Console UI itself. Please either isolate the changes into different PR's or open a new PR with all changes listed so that we can properly review the changes you're proposing without having to cross-reference your #7 as well. It also seems like there were some changes requested in #7 were not addressed in this PR as well.

everything from 7 should be in this PR and as for changes on 7 what where those changes not addressed? as i do remember fixing it all

DeathSoulYTDev commented:
i can also rename this PR and add a list of what was changed &  addedd

DeathSoulYTDev commented:
# PR Changes

- Added Support for Saving QC2 Tones
- Added Channel Sorting VIA Zone/Department Tabs,
- Filtered QC2 Tones Through Zone/Department Tabs to sit in same Tab and the Corresponding Channel
- Made QC2 Tone Boxes Draggable Along with Channel Boxes
- Redesigned The Console's Whole UI into a Dark Theme


> Thats what was added & changed

Firav commented:
> > @firealarmss My suggestion would be to close this PR and #7. @DeathSoulYTDev We need separate PR's made for the changed you're making. The changes outlined in this PR are not just adding support for multi-tone paging, but also large overhauls to the Console UI itself. Please either isolate the changes into different PR's or open a new PR with all changes listed so that we can properly review the changes you're proposing without having to cross-reference your #7 as well. It also seems like there were some changes requested in #7 were not addressed in this PR as well.
> 
> everything from 7 should be in this PR and as for changes on 7 what where those changes not addressed? as i do remember fixing it all

If thats the case, when using this PR as snapshot and attempting to load a codeplug with the `tones:` array results in a parsing error and crash. Works fine with the branch #7 is based on, just not this one.

> @firealarmss Is there a reason we couldn't just reuse the AlertTone box for this? That would allow us to keep all the same logic. You could use a property in the class to tell it if it is a tone box or not.

This was never addressed, you are still using new objects, not the AlertTone box.

DeathSoulYTDev committed:
Add zones to Station 1 and Station 2 in codeplug.yml

DeathSoulYTDev commented:
>> If thats the case, when using this PR as snapshot and attempting to load a codeplug with the `tones:` array results in a parsing error and crash. Works fine with the branch #7 is based on, just not this one.

> Fixed
 
> > @firealarmss Is there a reason we couldn't just reuse the AlertTone box for this? That would allow us to keep all the same logic. You could use a property in the class to tell it if it is a tone box or not.
>> This was never addressed, you are still using new objects, not the AlertTone box.

> was not able to find a way to make that work without major rewriting of the component and massive code base issues along side unreadable code so i had to stick with making it its own component

Firav commented:
> @DeathSoulYTDev was not able to fix a way to make that work with major rewriting of the component and mass code base issues along side unreadable code so i had to stick with making it its own component

@firealarmss back to you on this one in reference to your requested changes in #7
